logstash-core/src/main/java/org/logstash/FieldReference.java

```diff
@@ -126,7 +126,7 @@ public static void setEscapeStyle(final String escapeStyleSpec) {
      */
     private final int type;
 
-    private FieldReference(final String[] path, final String key, final int type) {
+    public FieldReference(final String[] path, final String key, final int type) {
```

Comment:
I am wary of making this `public`, since it would allow plugins built with the Java plugin API to route around our field reference caching.

Reply:
I revert it to private and build the FieldReference from string instead